src/renderer/components/+preferences/extensions.tsx

```diff
@@ -4,25 +4,29 @@
  */
 
 import { withInjectables } from "@ogre-tools/injectable-react";
-import type { IComputedValue } from "mobx";
 import { observer } from "mobx-react";
 import React from "react";
 import type { RegisteredAppPreference } from "./app-preferences/app-preference-registration";
+import extensionPreferencesModelInjectable from "./extension-preference-model.injectable";
 import { ExtensionSettings } from "./extension-settings";
 import { Preferences } from "./preferences";
-import extensionsPreferenceItemsInjectable from "./extension-preference-items.injectable";
 
 interface Dependencies {
-  preferenceItems: IComputedValue<RegisteredAppPreference[]>;
+  preferenceItems: RegisteredAppPreference[];
+  extensionName: string;
 }
 
-const NonInjectedExtensions = ({ preferenceItems }: Dependencies) => (
+const NonInjectedExtensions = ({ preferenceItems, extensionName }: Dependencies) => (
   <Preferences data-testid="extension-preferences-page">
     <section id="extensions">
-      <h2>Extensions</h2>
-      {preferenceItems.get().map((preferenceItem) => (
+      <h2>
+        {extensionName}
+        {" "}
+        preferences
+      </h2>
+      {preferenceItems.map((preferenceItem, index) => (
         <ExtensionSettings
-          key={preferenceItem.id}
+          key={`${preferenceItem.id}-${index}`}
```

Comment:
Is the preference item id not unique enough?

Reply:
ID should always be unique.